File: java/visitor/src/main/Payrise.java
package main;

public class Payrise extends VisitorBase{
    @Override
    public void Visit(Worker worker) {
        System.out.println(worker.Name + " salary increased to " + worker.Salary*1.1);
    }

    @Override
    public void Visit(Manager manager) {
        System.out.println(manager.Name + " salary increased to " + manager.Salary*1.2);
    }
}

File: csharp/Visitor/Payrise.cs
namespace Visitor;

public class Payrise : VisitorBase
{
    public override void Visit(Worker worker)
    {
        Console.WriteLine("{0} salary increased to {1}", worker.Name, worker.Salary*(decimal)1.1);
    }

    public override void Visit(Manager manager)
    {
        Console.WriteLine("{0} salary increased to {1}", manager.Name, manager.Salary*(decimal)1.2);
    }
}